medusa/essentials/dictionary.py

```diff
@@ -0,0 +1,28 @@
+# coding=utf-8
+"""Extended dictionaries."""
+
+from collections import OrderedDict
+
+
+def not_none(x):
+    """Return true is x is not none."""
+    return x is not None
+
+
+def not_empty(x):
+    """Return true is x is not none and not empty."""
+    return x is not None and x != ''
+
+
+class OrderedPredicateDict(OrderedDict):
+    """Ordered dictionary that only accept keys that matches the predicate."""
```

Comment:
Also instead of a predicate, the more pythonic way is to just subclass and extend the appropriate methods (e.g. setitem, missing, etc).